calabash-cucumber/lib/calabash-cucumber/store/preferences.rb

```diff
@@ -13,14 +14,80 @@ def initialize
         @path = File.join(dot_dir, "preferences", "preferences.json")
       end
 
+      # !@visibility private
+      def usage_tracking
+        preferences = read
+
+        unless valid_user_tracking_value?(preferences[:usage_tracking])
+          log_defaults_reset
+          preferences[:usage_tracking] = defaults[:usage_tracking]
+          write(preferences)
+        end
+
+        preferences[:usage_tracking]
+      end
+
+      # !@visibility private
+      def usage_tracking=(value)
+        if !valid_user_tracking_value?(value)
+          raise ArgumentError,
+            "Expected '#{value}' to be one of #{VALID_USAGE_TRACKING_VALUES.join(", ")}"
+        end
+
+        preferences = read
+        preferences[:usage_tracking] = value
+        write(preferences)
+      end
```

Comment:
I might have opted for a method like 
```ruby

def update_prefs(updates)
    preferences = read
    preferences.merge!(updates)
    write(preferences)
```

So that you could just do 
```ruby
update_prefs :user_tracking => value
```

But that's just syntactic style preference.